src/net/net.c

```diff
@@ -93,6 +101,21 @@ void lwip_deallocate(void *x)
     deallocate(lwip_heap, x, -1ull);
 }
 
+static boolean netif_ready(struct netif *n)
+{
+    return !ip_addr_isany(netif_ip_addr4(n));
+}
+
+static void check_netif_ready(struct netif *netif)
+{
+    if (netif_ready(netif)) {
+        remove_timer(kernel_timers, &net.timeout, 0);
+        status_handler c = pointer_from_u64(atomic_swap_64((u64 *)&net.ip4_complete, 0));
```

Comment:
Since cancelling a timer always invokes the timer handler, a further simplification would be to only invoke the status handler from net_timeout_handler, removing the check for ```overruns == timer_disabled```. Then this function only needs to call remove_timer() and nothing else, obviating the atomic swaps. The status handler may simply be stored in the timer handler closure.